Common/Data/FileSync.swift

```diff
@@ -0,0 +1,253 @@
+//
+//  Created for schulcloud-mobile-ios under GPL-3.0 license.
+//  Copyright © HPI. All rights reserved.
+//
+
+import BrightFutures
+import Marshal
+import Foundation
+
+public class FileSync: NSObject {
+
+    static public var `default`: FileSync = FileSync()
+
+    public typealias ProgressHandler = (Float) -> Void
+
+    fileprivate var backgroundSession: URLSession!
+    fileprivate var foregroundSession: URLSession!
+    fileprivate let metadataSession: URLSession
+
+    var runningTask: [Int: Promise<URL, SCError>] = [:]
+    var progressHandlers: [Int: ProgressHandler] = [:]
```

Comment:
use a tuple with `(Promise<URL, SCError>,  ProgressHandler)` and maybe also a typealias